src/lib/resource.js

```diff
@@ -1,51 +1,121 @@
-  }
-};
-
-export {
-  axios,
-  getResourceMetaDataJob,
-};
+/* eslint-disable import/prefer-default-export */
+import httpCodes from 'http-codes';
+import axios from 'axios';
+import { auditLogger, logger } from '../logger';
+import { Resource } from '../models';
+
+const getPageScrapeValues = async (resourceUrl, isEclkc) => {
+  // Use axios to get url info.
+  const res = await axios.get(resourceUrl, { maxRedirects: 3 });
+
+  // Scrape Title & National Centers from ECLKC resources.
+  const foundTitle = res.data.match(/<title[^>]*>([^<]+)<\/title>/);
+
+  // If we found a title update it.
+  if (foundTitle && foundTitle.length >= 1) {
+    // Get title.
+    const titleToUpdate = foundTitle[1].trim();
+
+    // Update db.
+    const updatedCnt = await Resource.update({
+      title: titleToUpdate,
+    }, {
+      where: { url: resourceUrl },
+      individualHooks: false,
+    });
+
+    // If we don't update anything throw an error to retry.
+    if (updatedCnt[0] === 0) {
+      throw Error('Failed to update db resource.');
+    }
+
+    return updatedCnt[0];
+  }
+
+  return 0;
+};
+
+const getMetadataValues = async (resourceId, resourceUrl) => {
+  // Attempt to get the resource metadata (if valid ECLKC resource).
+  // Sample: https://eclkc.ohs.acf.hhs.gov/mental-health/article/head-start-heals-campaign?_format=json
+  try {
+    const metadataUrl = `${resourceUrl}?_format=json`;
+    const res = await axios.get(metadataUrl, { maxRedirects: 3 });
+    const metadata = res.data;
+
+    // get created.
+    const { created } = metadata;
+    // get changed.
+    const { changed } = metadata;
+    // get title.
+    const { title } = metadata;
+    // get field_taxonomy_national_centers.
+    // eslint-disable-next-line max-len
+    const fieldTaxonomyNationalCenters = metadata.field_taxonomy_national_centers;
+    // get the field_taxonomy_topic.
+    const fieldTaxonomyTopic = metadata.field_taxonomy_topic;
+    // get the langcode.
+    const { langcode } = metadata;
+    // get the field_context.
+    const fieldContext = metadata.field_context;
+
+    // Create the metadata object.
+    const metadataObj = {
+      created,
+      changed,
+      title,
+      fieldTaxonomyNationalCenters,
+      fieldTaxonomyTopic,
+      langcode,
+      fieldContext,
+    };
+
+    // Update URL in DB.
+    await Resource.update({
+      metadata: metadataObj,
+      metadataUpdatedAt: new Date(),
+    }, {
+      where: { url: resourceUrl },
+      individualHooks: false,
+    });
+  } catch (error) {
+    auditLogger.error(`Resource Queue: Unable to retrieve ECLKC metadata for Resource (ID: ${resourceId} URL: ${resourceUrl}).`, error);
+  }
+};
+
+const getResourceMetaDataJob = async (job) => {
+  const {
+    resourceId, resourceUrl,
+  } = job.data;
+
+  try {
+    // 1.) Determine if this is an ECLKC resource.
+    const isEclkc = resourceUrl.includes('eclkc.ohs.acf.hhs.gov');
+
+    // 2.) Scrape page title for all pages.

```

Comment:
if it's ECLKC, why not get the title w/the metadata so we don't have to scrape in that case?

Reply:
I want to keep the title as is it also triggers a retry.

Reply:
We also might wind up scraping the national centers in the future.

Reply:
So we are going to fetch the title with the metadata and then fetch it again with the scrape? 
Not sure how the retry factors in - it wouldn't need to retry if it already had the title, right?

Reply:
Couple of things:

- When I test this out, it throws an error in the console despite successfully fetching the metadata. Probably a simple fix.
- There are ECLKC pages without a JSON response. The way this function is written, those pages would not have their titles retrieved.
- I think your code also bombs if there is no title but the other metadata is valid (an unlikely scenario)

I created [a PR to this branch](https://github.com/HHS/Head-Start-TTADP/pull/1581) demonstrating how I imagined this would work. Try once to get the metadata, if it fails, then we try the page title, if that fails, we go to the retry loop. 

I also incorporated Jon's idea, which I think it probably correct, that we should just save the entire metadata object, which simplifies the code for the metadata scrape.